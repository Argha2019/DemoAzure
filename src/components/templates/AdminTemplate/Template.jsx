import * as React from "react";
import Card from "./../../atoms/card/card.jsx";
import Header from "./../../molecules/header/Admin.jsx";
import Footer from "./../../molecules/footer/footer.jsx";
import "../../../css/global.css";
import axios from "axios";
import Cookies from "js-cookie";
import { Button } from "react-bootstrap";
import { encryptByAES,decryptByAES } from "../../pages/Encryption.jsx";
import Config from "../../../config/urls.jsx";


/**
 * @author kasjain(Kashish Jain)
 * This file is responsible for making the header and footer template for role:"ADMIN"
 * which is always rendered in every page.
 */

class AdminTemplate extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			user: "loading"
		};
	}

	componentDidMount = () => {
		if (Cookies.get("adminData") === undefined) {
			this.setState({ user: false });
			return;
		} else {
			this.setState({ user: true });
			var url = Config.serverUrl + '/signin/isvalid';
			try {

				axios.post(url,'"' + encryptByAES(JSON.parse(decryptByAES(Cookies.get('adminData')))[0].accessToken) + '"', {headers: {'Content-Type': 'application/json'}})

					.then(res => {
						if (res.data) {
							this.setState({ user: true });
						} else {
							Cookies.remove("adminData");
							this.setState({ user: false });
						}
					});
			} catch (e) {
				console.log("Request not found!!");
			}
		}
	};

	render() {
		if (this.state.user == true) {
			return (
				<Card className={this.props.className}>
					<Header></Header>
					{this.props.children}
					<Footer></Footer>
				</Card>
			);
		} else if (this.state.user == false) {
			return (
				<div>
					<br/>
					<p align="center">You must Login!!</p>
					<Button className="editButtonDiv" variant="primary" href="#/">
						Sign In
					</Button>
				</div>
			);
		} else {
			return <div />;
		}
	}
}

export default AdminTemplate;
