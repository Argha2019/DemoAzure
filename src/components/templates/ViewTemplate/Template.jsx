import * as React from "react";
import Card from "./../../atoms/card/card.jsx";
import Header from "./../../molecules/header/AgentView.jsx";
import Footer from "./../../molecules/footer/footer.jsx";
import "../../../css/global.css";
import axios from "axios";
import Cookies from "js-cookie";
import { Button } from "react-bootstrap";
import { encryptByAES,decryptByAES } from "../../pages/Encryption.jsx";
import Config from "../../../config/urls.jsx";


/**
 * @author Arghadeep Mandal
 * This file is responsible for making the header and footer template for role:"VIEW"
 * which is always rendered in every page.
 */

class ViewTemplate extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			user: "loading"
		};
	}

	componentDidMount = () => {
		if (Cookies.get("viewData") === undefined) {
			this.setState({ user: false });
			return;
		} else {
			this.setState({ user: true });
			try {
				var url = Config.serverUrl + '/signin/isvalid';
				axios.post(url ,'"' + encryptByAES(JSON.parse(decryptByAES(Cookies.get('viewData')))[0].accessToken) + '"', {headers: {'Content-Type': 'application/json'}})
					.then(res => {
						if (res.data) {
							this.setState({ user: true });
						} else {
							Cookies.remove("viewData");
							this.setState({ user: false });
						}
					});
			} catch (e) {
				console.log("Request not found!!");
			}
		}
	};

	render() {
		if (this.state.user == true) {
			return (
				<Card className={this.props.className}>
					<Header></Header>
					{this.props.children}
					<Footer></Footer>
				</Card>
			);
		} else if (this.state.user == false) {
			return (
				<div>
					<br/>
					<p align="center">You must Login!!</p>
					<Button className="editButtonDiv"  variant="primary" href="#/">
						Sign In
					</Button>
				</div>
			);
		} else {
			return <div />;
		}
	}
}

export default ViewTemplate;
