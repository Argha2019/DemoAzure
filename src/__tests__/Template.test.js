import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import AdminTemplate from '../components/templates/AdminTemplate/Template.jsx';
import AgentTemplate from '../components/templates/AgentTemplate/Template.jsx';
import ViewTemplate from '../components/templates/ViewTemplate/Template.jsx';
import CustomerTemplate from '../components/templates/CustomerTemplate/Template.jsx';

import axios from 'axios';
import Cookies from 'js-cookie';

jest.mock('axios');

describe('Admin Template', () => {


	it('should render snapshot', () => {
		const resp = { data : "kjdcsuhdjdjc"};
		axios.post.mockImplementation(() => Promise.resolve({ status: 201 , data: {accessToken: "asdfdsdfddgfdgfd"} }));
		let wrapper = shallow(<AdminTemplate />);
		
		Cookies.set('adminData', "Amit");
		expect(toJson(wrapper)).toMatchSnapshot();

		wrapper = shallow(<AdminTemplate />);

		expect(wrapper.state('user')).toBe(true);
		
		Cookies.remove('adminData');
		
		
	});
    
});

describe('Agent Template' , () => {
	it('should render snapshot', () => {
		const resp = { data : "kjdcsuhdjdjc"};
		axios.post.mockImplementation(() => Promise.resolve({ status: 201 , data: {accessToken: "asdfdsdfddgfdgfd"} }));
		let wrapper = shallow(<AgentTemplate />);
		
		Cookies.set('agentData', "Amit");
		expect(toJson(wrapper)).toMatchSnapshot();

		wrapper = shallow(<AgentTemplate />);

		expect(wrapper.state('user')).toBe(true);
		
		Cookies.remove('agentData');
		
		
	});
});

describe('View Template' , () => {
	it('should render snapshot', () => {
		const resp = { data : "kjdcsuhdjdjc"};
		axios.post.mockImplementation(() => Promise.resolve({ status: 201 , data: {accessToken: "asdfdsdfddgfdgfd"} }));
		let wrapper = shallow(<ViewTemplate />);
		
		Cookies.set('viewData', "Amit");
		expect(toJson(wrapper)).toMatchSnapshot();

		wrapper = shallow(<ViewTemplate />);

		expect(wrapper.state('user')).toBe(true);
		
		Cookies.remove('viewData');
		
		
	});
});

describe('Customer Template' , () => {
	it('should render snapshot', () => {
		const resp = { data : "kjdcsuhdjdjc"};
		axios.post.mockImplementation(() => Promise.resolve({ status: 201 , data: {accessToken: "asdfdsdfddgfdgfd"} }));
		let wrapper = shallow(<CustomerTemplate />);
		
		Cookies.set('customerData', "Amit");
		expect(toJson(wrapper)).toMatchSnapshot();

		wrapper = shallow(<CustomerTemplate />);

		expect(wrapper.state('user')).toBe(true);
		
		Cookies.remove('customerData');
		
		
	});
});

