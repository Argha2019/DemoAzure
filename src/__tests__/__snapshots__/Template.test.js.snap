// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Template should render snapshot 1`] = `
<div>
  <br />
  <p
    align="center"
  >
    You must Login!!
  </p>
  <Button
    active={false}
    className="editButtonDiv"
    disabled={false}
    href="#/"
    type="button"
    variant="primary"
  >
    Sign In
  </Button>
</div>
`;

exports[`Agent Template should render snapshot 1`] = `
<div>
  <br />
  <p
    align="center"
  >
    You must Login!!
  </p>
  <Button
    active={false}
    className="editButtonDiv"
    disabled={false}
    href="#/"
    type="button"
    variant="primary"
  >
    Sign In
  </Button>
</div>
`;

exports[`Customer Template should render snapshot 1`] = `
<div>
  <br />
  <p
    align="center"
  >
    You must Login!!
  </p>
  <Button
    active={false}
    className="editButtonDiv"
    disabled={false}
    href="#/"
    type="button"
    variant="primary"
  >
    Sign In
  </Button>
</div>
`;

exports[`View Template should render snapshot 1`] = `
<div>
  <br />
  <p
    align="center"
  >
    You must Login!!
  </p>
  <Button
    active={false}
    className="editButtonDiv"
    disabled={false}
    href="#/"
    type="button"
    variant="primary"
  >
    Sign In
  </Button>
</div>
`;
